tools/changelogger/src/render.rs

```diff
@@ -275,56 +288,125 @@ fn update_changelogs(
         args.previous_release_versions_manifest.as_deref(),
     )?;
-    let (release_header, release_notes) = render(&entries, &release_metadata.title);
+    match args.change_set {
+        ChangeSet::AwsSdk => render_aws_rust(args, &entries, &aws_sdk_rust_metadata),
+        ChangeSet::SmithyRs => render_smithy_rs(args, &entries, &smithy_rs_metadata)
+    }?;
+    std::fs::write(&args.source_to_truncate, EXAMPLE_ENTRY.trim())
+        .context("failed to truncate source")?;
+    eprintln!("Changelogs updated!");
+    Ok(())
+}
+
+fn render_aws_rust(args : &RenderArgs, entries : &[ChangelogEntry], release_metadata : &ReleaseMetadata) -> Result<()> {
+    let (release_header, release_notes) = render(entries, &release_metadata.title, false);
     if let Some(output_path) = &args.release_manifest_output {
-        let release_manifest = ReleaseManifest {
-            tag_name: release_metadata.tag.clone(),
-            name: release_metadata.title.clone(),
-            body: release_notes.clone(),
-            // All releases are pre-releases for now
-            prerelease: true,
-        };
+        write_release_manifest(output_path, release_metadata, &release_notes)?
+    }
+    let _ = write_changelog_md(&release_header, &release_notes, &args.changelog_output)?;
+    Ok(())
+}
+
+fn render_smithy_rs(args: &RenderArgs, entries: &[ChangelogEntry], release_metadata: &ReleaseMetadata) -> Result<()> {
+    let server_output_path = args.server_changelog_output.as_ref()
+            .ok_or_else(|| anyhow::Error::msg(format!("server sdk output path has not been supplied. Please use --server-output-path as parameter")))?;
+
+    if let Some(output_path) = &args.release_manifest_output {
+        let (_, combined_release_notes) = render(&entries, &release_metadata.title, true);
+        write_release_manifest(output_path, release_metadata, &combined_release_notes)?;
+    }
+    
+    let (client_entries, server_entries) = partition_entries(entries);
+    let (release_header, release_notes) = render(&client_entries, &release_metadata.title, false);
+    let (_, server_release_notes) = render(&server_entries, &release_metadata.title, false);
+
+    let old_client_contents = write_changelog_md(&release_header, &release_notes, &args.changelog_output)?;
+
+    if let Err(e) = write_changelog_md(&release_header, &server_release_notes, &server_output_path) {
+        // restore client sdk changelog back to its original state
         std::fs::write(
-            output_path.join(&release_metadata.manifest_name),
-            serde_json::to_string_pretty(&release_manifest)?,
+            &args.changelog_output,
+            old_client_contents
         )
-        .context("failed to write release manifest")?;
+        .context(format!("server changelog output could not be written (error: {e}). Failed to restore client changelog!"))?;
+
+        Err(anyhow::Error::msg(format!("Could not write server release notes. Error: {e}")))
+    }
```

Comment:
This tool is basically always run in automation, so you can just `?` operator the error on the second `write_changelog_md` since that will fail the whole release GitHub Actions workflow, which means it won't matter that the client file was overwritten since that whole workspace gets thrown away.

If someone did run it manually, they can use git to restore files.